tests/parser/syntax/test_extract32.py

```diff
@@ -1,23 +1,57 @@
 import pytest
-from pytest import raises
 
-from vyper import compiler
-from vyper.exceptions import TypeMismatch
+from vyper.exceptions import InvalidType, TypeMismatch
 
 fail_list = [
-    """
+    (
+        """
 @external
 def foo() -> uint256:
     return extract32(b"cowcowcowcowcowccowcowcowcowcowccowcowcowcowcowccowcowcowcowcowc", 0)
-    """
+    """,
+        TypeMismatch,
+    ),
+    (
+        """
+@external
+def foo(inp: address) -> int128:
```

Comment:
might be good add a brief comment to each of these saying why it should throw an exception. ex.
```
# can't extract32 on an address
```